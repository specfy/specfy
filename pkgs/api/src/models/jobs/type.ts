import type { Prisma } from '@prisma/client';

export interface JobDeployConfig {
  url: string;
  autoLayout?: boolean;
  hook?: {
    id: string;
    ref: string;
  };
}

export type JobWithOrgProject = Prisma.JobsGetPayload<{
  include: { Org: true; Project: true };
}>;

export type JobCode =
  | 'cant_auth_github'
  | 'failed_to_checkout'
  | 'failed_to_cleanup'
  | 'failed_to_clone'
  | 'failed_to_deploy'
  | 'failed_to_start_github_deployment'
  | 'failed_to_teardown'
  | 'missing_dependencies'
  | 'no_api_key'
  | 'org_not_installed'
  | 'project_not_installed'
  | 'success'
  | 'unknown';
