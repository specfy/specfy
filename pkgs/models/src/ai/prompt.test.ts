import { describe, expect, it } from 'vitest';

import {
  aiPromptProjectDescription,
  aiPromptRewrite,
  aiPromptProjectOnboarding,
} from './prompt.js';

import type { ComponentForPrompt } from './types.js';

const components: ComponentForPrompt[] = [
  { id: '1', name: 'GCP', type: 'cloud', techId: 'gcp', slug: 'gcp' },
  { id: '2', name: 'GCE', type: 'hosting', techId: 'gcp.gce', slug: 'gce' },
  { id: '3', name: 'Aggregator', type: 'service', techId: null, slug: 'aggr' },
  {
    id: '4',
    name: 'Algolia',
    type: 'saas',
    techId: 'algolia',
    slug: 'algolia',
  },
  {
    id: '5',
    name: 'Postgres',
    type: 'db',
    techId: 'postgres',
    slug: 'postgres',
  },
  { id: '6', name: 'SQS', type: 'messaging', techId: 'aws.sqs', slug: 'sqs' },
  { id: '7', name: 'Billing', type: 'project', techId: null, slug: 'billing' },
];

describe('aiPromptRewrite', () => {
  it('should output the prompt', () => {
    const res = aiPromptRewrite({
      text: 'Specfy is an awesome platform',
    });
    expect(res).toMatchSnapshot();
  });
});

describe('aiPromptProjectDescription', () => {
  it('should output the prompt', () => {
    const res = aiPromptProjectDescription({
      project: { name: 'Analytics' },
      components,
    });
    expect(res).toMatchSnapshot();
  });
});

describe('aiPromptProjectOnboarding', () => {
  it('should output the prompt', () => {
    const res = aiPromptProjectOnboarding({
      project: {
        name: 'Analytics',
        slug: 'analytics',
        orgId: 'acme',
        links: [{ title: 'GitHub', url: 'https://github.com/specfy/specfy' }],
        description: {
          type: 'doc',
          content: [
            {
              type: 'paragraph',
              attrs: { uid: 'UidC3Ls190' },
              content: [
                {
                  type: 'text',
                  text: 'The Analytics project collects data from various sources, processes it, and presents it in dashboards and reports. These outputs show key indicators and trends for the business.',
                },
                { type: 'hardBreak' },
                {
                  type: 'text',
                  text: 'The project also employs machine learning and statistical methods to analyze data. This analysis helps in predicting trends and making recommendations based on past data. This project assists businesses in making data-based decisions.',
                },
              ],
            },
          ],
        },
      },
      components,
      documents: [
        { name: 'README', slug: 'readme' },
        { name: 'Contributing', slug: 'docs/contributing' },
        { name: 'Installation', slug: 'docs/installation' },
        { name: 'Deploy', slug: 'docs/deploy' },
        { name: 'Deploy With Terraform', slug: 'docs/deploy.terraform' },
      ],
    });
    expect(res).toMatchSnapshot();
  });
});
