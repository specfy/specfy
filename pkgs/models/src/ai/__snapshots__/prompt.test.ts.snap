// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`aiPromptProjectDescription > should output the prompt 1`] = `
[
  {
    "content": "You describe a technical project for an engineer.
3 paragraphs maximum
DO NOT use adjective, superlative or adverb.
Details why it exists in one paragraph with a line break at end
Details briefly how it works
Use the list of technologies to describe how it works. Choose only the most relevant ones",
    "role": "system",
  },
  {
    "content": "Project name is Analytics

Technologies:
Cloud: GCP
Hosting: GCE
Service: Aggregator
Saas: Algolia
Database: Postgres
Messaging: SQS
Depends on internal product: Billing
  ",
    "role": "user",
  },
]
`;

exports[`aiPromptProjectOnboarding > should output the prompt 1`] = `
[
  {
    "content": "You are technical writer. With the information provided write a README.md that you would find in GitHub.
Split in 4 sections:
1. An Overview section
2. A Get Started section, bullet point sentences with one emoji per line. Use the list of documents provided, ONLY use what you would find in a README (e.g: installation, guidelines, deployment, contributing, etc.)
3. A Technology section, highlighting 3-5 relevant technologies in the list, with a quick sentence how about what they do and the link
4. A next step section, write use generic advices and use the links in the last list

If the provided information needed to write a section are not enough then simply write: \\"Insufficient information.\\". The writer should refrain using links that were not provided by the user.
End the content with encouraging words with emojis.
Output in markdown",
    "role": "system",
  },
  {
    "content": "Project name is Analytics
The Analytics project collects data from various sources, processes it, and presents it in dashboards and reports. These outputs show key indicators and trends for the business.

Technologies:
Hosting: GCP http://localhost:5173/acme/analytics/c/undefined-gcp, GCE http://localhost:5173/acme/analytics/c/undefined-gce
Database: Postgres http://localhost:5173/acme/analytics/c/undefined-postgres
Depends on internal product: Billing http://localhost:5173/acme/analytics/c/undefined-billing

Documentation:
http://localhost:5173/acme/analytics/doc/readme: README
http://localhost:5173/acme/analytics/doc/docs/contributing: Contributing
http://localhost:5173/acme/analytics/doc/docs/installation: Installation
http://localhost:5173/acme/analytics/doc/docs/deploy: Deploy
http://localhost:5173/acme/analytics/doc/docs/deploy.terraform: Deploy With Terraform
http://localhost:5173/acme/analytics/doc/docs/foobar: Foobar

Links:
GitHub: https://github.com/specfy/specfy
Discord: https://discord.gg/foobar
Infrastructure Flow: http://localhost:5173/acme/analytics/flow",
    "role": "user",
  },
  {
    "content": "# Overview
The Analytics project collects data from various sources, processes it, and presents it in dashboards and reports. These outputs show key indicators and trends for the business. The project also employs machine learning and statistical methods to analyze data, helping businesses make data-based decisions.

## Get Started
- ðŸ“¥ **Installation**: Clone the repository from [GitHub](https://github.com/specfy/specfy)

## Technology
The Analytics project utilizes the following technologies:
- The project is hosted on [GCP](http://localhost:5173/acme/analytics/c/jZDC3Lsc01-gcp)
- We are using [Postgres](http://localhost:5173/acme/analytics/c/jZDC3Lsc04-postgresql) as the main Database
- You would want to checkout out [Billing](http://localhost:5173/acme/analytics/c/undefined-billing) project.

## Next Steps
To further explore the project, you can:
- Check the [GitHub repository](https://github.com/specfy/specfy)
- Join our community on the [Discord](https://discord.gg/foobar)
- Follow the infrastructure flow [here](http://localhost:5173/acme/analytics/flow) to understand the project's architecture.",
    "role": "assistant",
  },
  {
    "content": "Project name is Analytics
The Analytics project collects data from various sources, processes it, and presents it in dashboards and reports. These outputs show key indicators and trends for the business.
The project also employs machine learning and statistical methods to analyze data. This analysis helps in predicting trends and making recommendations based on past data. This project assists businesses in making data-based decisions.



Technologies:
Cloud: GCP http://localhost:5173/acme/analytics/c/1-gcp
Hosting: GCE http://localhost:5173/acme/analytics/c/2-gce
Service: Aggregator http://localhost:5173/acme/analytics/c/3-aggr
Saas: Algolia http://localhost:5173/acme/analytics/c/4-algolia
Database: Postgres http://localhost:5173/acme/analytics/c/5-postgres
Messaging: SQS http://localhost:5173/acme/analytics/c/6-sqs
Depends on internal product: Billing http://localhost:5173/acme/analytics/c/7-billing

Documentation:
http://localhost:5173/acme/analytics/doc/readme: README
http://localhost:5173/acme/analytics/doc/docs/contributing: Contributing
http://localhost:5173/acme/analytics/doc/docs/installation: Installation
http://localhost:5173/acme/analytics/doc/docs/deploy: Deploy
http://localhost:5173/acme/analytics/doc/docs/deploy.terraform: Deploy With Terraform

Links:
GitHub: https://github.com/specfy/specfy
Infrastructure Flow: http://localhost:5173/acme/analytics/flow
",
    "role": "user",
  },
]
`;

exports[`aiPromptRewrite > should output the prompt 1`] = `
[
  {
    "content": "You rewrite content you are provided with for an engineer. Do not use adverb or subjective language.",
    "role": "system",
  },
  {
    "content": "Specfy is an awesome platform",
    "role": "user",
  },
]
`;
